graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/functions/lookup/LookupAddStringList.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.plugins.pipelineprocessor.functions.lookup;
+
+import com.google.inject.Inject;
+import org.graylog.plugins.pipelineprocessor.EvaluationContext;
+import org.graylog.plugins.pipelineprocessor.ast.functions.AbstractFunction;
+import org.graylog.plugins.pipelineprocessor.ast.functions.FunctionArgs;
+import org.graylog.plugins.pipelineprocessor.ast.functions.FunctionDescriptor;
+import org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor;
+import org.graylog2.lookup.LookupTableService;
+
+import java.util.List;
+
+import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.bool;
+import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.object;
+import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.string;
+
+public class LookupAddStringList extends AbstractFunction<Object> {
+
+    public static final String NAME = "lookup_add_string_list";
+
+    private final ParameterDescriptor<String, LookupTableService.Function> lookupTableParam;
+    private final ParameterDescriptor<Object, Object> keyParam;
+    @SuppressWarnings("rawtypes")
+    private final ParameterDescriptor<List, List> valueParam;
+    private final ParameterDescriptor<Boolean, Boolean> appendOption;
+
+    @Inject
+    public LookupAddStringList(LookupTableService lookupTableService) {
+        lookupTableParam = string("lookup_table", LookupTableService.Function.class)
+                .description("The existing lookup table to use to add the given list")
+                .transform(tableName -> lookupTableService.newBuilder().lookupTable(tableName).build())
+                .build();
+        keyParam = object("key")
+                .description("The key to add in the lookup table")
+                .build();
+        valueParam = ParameterDescriptor.type("list_value", List.class)
+                .description("The list value that should be added into the lookup table")
+                .build();
+        appendOption = bool("append_values")
```

Comment:
What do you think about naming this (and also in other parts of the backend code) `keep_duplicates`? From reading the parameter name, I wasn't really sure what it meant.